tests/common/ixia/common_helpers.py

```diff
@@ -200,11 +200,182 @@ def pfc_class_enable_vector(prio_list):
+                u'wred_red_enable': u'true',
+                u'wred_yellow_enable': u'true',
+                u'yellow_drop_probability': u'5',
+                u'yellow_max_threshold': u'2097152',
+                u'yellow_min_threshold': u'1048576'
+            }
+        }
+    """
+    config_facts = host_ans.config_facts(host=host_ans.hostname,
+                                         source="running")['ansible_facts']
+
+    if "WRED_PROFILE" in config_facts.keys():
+        return config_facts['WRED_PROFILE']
+    else:
+        return None
+
+def config_wred(host_ans, kmin, kmax, pmax, profile=None):
+    """
+    Config a WRED/ECN profile of a SONiC switch
+
+    Args:
+        host_ans: Ansible host instance of the device
+        kmin (int): RED/ECN minimum threshold in bytes
+        kmax (int): RED/ECN maximum threshold in bytes
+        pmax (int): RED/ECN maximum marking probability in percentage
+        profile (str): name of profile to configure (None means any profile)
+
+    Returns:
+        If configuration succeeds (bool)
+    """
+
+    if not isinstance(kmin, int) or \
+       not isinstance(kmax, int) or \
+       not isinstance(pmax, int):
+        return False
+
+    if kmin < 0 or kmax < 0 or pmax < 0 or pmax > 100 or kmin > kmax:
+        return False
+
+    profiles = get_wred_profiles(host_ans)
+    """ Cannot find any WRED/ECN profiles """
+    if profiles is None:
+        return False
+
+    """ Cannot find the profile to configure at the device """
+    if profile is not None and profile not in profiles:
+        return False
+
+    for p in profiles:
+        """ This is not the profile to configure """
+        if profile is not None and profile != p:
+            continue
+
+        kmin_old = int(profiles[p]['green_min_threshold'])
+        kmax_old = int(profiles[p]['green_max_threshold'])
+
+        if kmin_old > kmax_old:
+            return False
+
+        """ Ensure that Kmin is no larger than Kmax during the update """
+        if kmin > kmin_old:
+            host_ans.shell('sudo ecnconfig -p {} -gmax {}'.format(p, kmax))
+            host_ans.shell('sudo ecnconfig -p {} -gmin {}'.format(p, kmin))
+
+        else:
+            host_ans.shell('sudo ecnconfig -p {} -gmin {}'.format(p, kmin))
+            host_ans.shell('sudo ecnconfig -p {} -gmax {}'.format(p, kmax))
+
+    return True
+
+def enable_ecn(host_ans, prio):
+    """
+    Enable ECN marking on a priority
+
+    Args:
+        host_ans: Ansible host instance of the device
+        prio (int): priority
+
+    Returns:
+        N/A
+    """
+    host_ans.shell('sudo ecnconfig -q {} on'.format(prio))
+
+def disable_ecn(host_ans, prio):
+    """
+    Disable ECN marking on a priority
+
+    Args:
+        host_ans: Ansible host instance of the device
+        prio (int): priority
+
+    Returns:
+        N/A
+    """
+    host_ans.shell('sudo ecnconfig -q {} off'.format(prio))
+
+def config_buffer(host_ans, profile, alpha_log2):
```

Comment:
can you rename to state that config done is for alpha?

Reply:
Fixed